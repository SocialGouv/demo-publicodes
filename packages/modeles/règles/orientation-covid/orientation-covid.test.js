/* eslint-env node18 */
import fs from "fs";
import yaml from "js-yaml";
import path from "path";
import Engine from "publicodes";

const fullpath = path.join("règles", "orientation-covid", "publicodes.yaml");
const rules = yaml.load(fs.readFileSync(fullpath, "utf-8"));

test("a besoin de variables pour fonctionner", () => {
  const engine = new Engine(rules);
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.missingVariables).toMatchSnapshot();
});

const defaultSituation = {
  "patient . âge": 25,
  "patient . antécédents de maladie cardiovasculaire": "non",
  "patient . cancert évolutif": "non",
  "patient . cirrhose": "non",
  "patient . diabète mal équilibré ou avec des complications": "non",
  "patient . drépanocytose": "non",
  "patient . enceinte": "non",
  "patient . immunodépression": "non",
  "patient . insuffisance rénale": "non",
  "patient . maladie respiratoire chronique": "non",
  "patient . poids": 80,
  "patient . taille": 180,
  "symptômes . alimenter ou boire impossible": "non",
  "symptômes . anosmie": "non",
  "symptômes . diarrhée": "non",
  "symptômes . douleurs": "non",
  "symptômes . fatigue": "non",
  "symptômes . manque de souffle": "non",
  "symptômes . température": 38,
  "symptômes . toux": "non",
};

test("ne renvoie pas d'alerte avec la situation par défaut", () => {
  const engine = new Engine(rules);
  engine.setSituation(defaultSituation);
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.nodeValue).toMatchSnapshot();
  expect(evaluated.missingVariables).toEqual({});
});

const tests = [
  {
    title: "enfants < 15 ans",
    situation: {
      "patient . âge": 12,
    },
  },
  {
    title: "facteurs de gravité majeure",
    situation: {
      "patient . âge": 22,
      "symptômes . manque de souffle": "oui", // facteur de gravité majeure
    },
  },
  {
    title: "fièvre 25 ans",
    situation: {
      "patient . âge": 25,
      "symptômes . température": 40, // facteur de gravité mineure
    },
  },
  {
    title: "fièvre 55 ans",
    situation: {
      "patient . âge": 55,
      "symptômes . température": 40, // facteur de gravité mineure
    },
  },
  {
    title: "fièvre 70 ans",
    situation: {
      "patient . âge": 70, // facteur pronostique
      "symptômes . température": 40, // facteur de gravité mineure
    },
  },
  {
    title: "fièvre et toux",
    situation: {
      "symptômes . température": 40, // facteur de gravité mineure
      "symptômes . toux": "oui",
    },
  },
  {
    title: "fièvre et toux et cirrhose",
    situation: {
      "symptômes . température": 40, // facteur de gravité mineure
      "symptômes . toux": "oui",
      "patient . cirrhose": "oui", // facteur pronostique
    },
  },
  {
    title: "fièvre et toux et cirrhose et fatigue importante",
    situation: {
      "symptômes . température": 40, // facteur de gravité mineur
      "symptômes . toux": "oui",
      "symptômes . fatigue": "oui",
      "symptômes . fatigue importante": "oui", // facteur de gravité mineur
      "patient . cirrhose": "oui", // facteur pronostique
    },
  },
  {
    title: "38 ans et toux",
    situation: {
      "symptômes . température": 38,
      "symptômes . toux": "oui",
    },
  },
  {
    title: "38 ans et toux et facteur pronostique mineur",
    situation: {
      "symptômes . température": 38,
      "symptômes . toux": "oui",
      "patient . cirrhose": "oui", // facteur de granité mineur
    },
  },
];

test.each(tests)("$title", ({ title, situation }) => {
  const engine = new Engine(rules);
  engine.setSituation({
    ...defaultSituation,
    ...situation,
  });
  const evaluated = engine.evaluate("résultat");

  expect(evaluated.nodeValue).toMatchSnapshot();
  expect(evaluated.missingVariables).toEqual({});
});
