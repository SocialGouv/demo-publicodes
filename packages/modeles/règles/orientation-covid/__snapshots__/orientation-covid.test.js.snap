// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a besoin de variables pour fonctionner 1`] = `
{
  "patient . antécédents de maladie cardiovasculaire": 127,
  "patient . cancert évolutif": 127,
  "patient . cirrhose": 127,
  "patient . diabète mal équilibré ou avec des complications": 127,
  "patient . drépanocytose": 127,
  "patient . enceinte": 127,
  "patient . immunodépression": 127,
  "patient . insuffisance rénale": 127,
  "patient . maladie respiratoire chronique": 127,
  "patient . poids": 127,
  "patient . taille": 163,
  "patient . âge": 172,
  "symptômes . alimenter ou boire impossible": 16,
  "symptômes . anosmie": 37,
  "symptômes . diarrhée": 16,
  "symptômes . douleurs": 37,
  "symptômes . fatigue": 43,
  "symptômes . manque de souffle": 16,
  "symptômes . température": 143,
  "symptômes . toux": 52,
}
`;

exports[`enfants < 15 ans 1`] = `
"Cette application n’est pas faite pour les personnes de moins de 15 ans.
Prenez contact avec votre médecin généraliste au moindre doute.

En cas d’urgence, appelez le 15.
"
`;

exports[`facteurs de gravité majeure 1`] = `
"Appelez le 15.
"
`;

exports[`fièvre 25 ans 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite
à domicile (SOS médecins, etc.)

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`fièvre 55 ans 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite à domicile.

Appelez le 15 si une gêne respiratoire ou des difficultés importantes
pour vous alimenter ou boire apparaissent pendant plus de 24 heures.

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`fièvre 70 ans 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite à domicile.

Appelez le 15 si une gêne respiratoire ou des difficultés importantes
pour vous alimenter ou boire apparaissent pendant plus de 24 heures.

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`fièvre et toux 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite
à domicile (SOS médecins, etc.)

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`fièvre et toux et cirrhose 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite à domicile.

Appelez le 15 si une gêne respiratoire ou des difficultés importantes
pour vous alimenter ou boire apparaissent pendant plus de 24 heures.

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`fièvre et toux et cirrhose et fatigue importante 1`] = `
"Votre situation peut relever d’un Covid-19.

Demandez une téléconsultation ou un médecin généraliste ou une visite
à domicile. Si vous n'arrivez pas à obtenir de consultation, appelez
le 15.

En attendant et pour casser les chaînes de transmission, nous vous
conseillons de vous isoler et de respecter les gestes barrières pour
protéger vos proches.
"
`;

exports[`ne renvoie pas d'alerte avec la situation par défaut 1`] = `
"Votre situation ne relève probablement pas du Covid-19.

N’hésitez pas à contacter votre médecin en cas de doute.

Vous pouvez refaire le test en cas de nouveau symptôme pour réévaluer
la situation.

Pour toute information concernant le Covid-19, composer le 0 800 130 000.
"
`;
